Source/WebCore/Modules/plugins/YouTubePluginReplacement.cpp

```diff
@@ -76,16 +76,20 @@ RenderPtr<RenderElement> YouTubePluginReplacement::createElementRenderer(HTMLPlu
     if (!m_embedShadowElement)
         return nullptr;
     
-    return m_embedShadowElement->createElementRenderer(WTFMove(style), insertionPosition);
+    return Ref { *m_embedShadowElement }->createElementRenderer(WTFMove(style), insertionPosition);
 }
 
 void YouTubePluginReplacement::installReplacement(ShadowRoot& root)
 {
-    m_embedShadowElement = YouTubeEmbedShadowElement::create(m_parentElement->document());
+    Ref document = m_parentElement->document();
 
-    root.appendChild(*m_embedShadowElement);
+    m_embedShadowElement = YouTubeEmbedShadowElement::create(document);
 
-    auto iframeElement = HTMLIFrameElement::create(HTMLNames::iframeTag, m_parentElement->document());
+    Ref embedShadowElement = *m_embedShadowElement;
+
+    root.appendChild(embedShadowElement);
+
+    auto iframeElement = HTMLIFrameElement::create(HTMLNames::iframeTag, document);
```

Comment:
`auto` -> `Ref` as we don't like to hide smart pointers behind auto in new code.